src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,175 @@
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get info from cars, manufacturers dbs", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT drivers.name, drivers.license_number,"
+                + " manufacturers.name as manufacturer, manufacturers.country, "
+                + "cars.id, cars.model "
+                + "FROM  cars_drivers "
+                + "INNER JOIN cars on cars_drivers.car_id = cars.id "
+                + "inner JOIN manufacturers on cars.manufacturer_id = manufacturers.id "
+                + "INNER JOIN drivers on cars_drivers.driver_id = drivers.id "
+                + "WHERE driver_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(createNewCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("can not get a list of cars of driver. "
+                    + "Params: id=" + driverId, e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE cars.id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturerId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not update car. Params: "
+                    + "id=" + car.getId()
+                    + ", model=" + car.getModel()
+                    + ", manufacturer_id=" + car.getManufacturerId(), e);
+        }
+        addDriverToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = true "
+                + "WHERE cars.id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            statement.executeUpdate();
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not delete car by id. Params: "
+                    + "id=" + id, e);
+        }
+    }
+
+    private Car createNewCar(ResultSet resultSet) throws SQLException {
```

Comment:
```suggestion
    private Car parseCar(ResultSet resultSet) throws SQLException {
```
or getCarFromResultSet